include/base/factory.h

```diff
@@ -146,7 +146,7 @@ std::unique_ptr<Base> Factory<Base>::build (const std::string & name)
     }
 
   Factory<Base> * f = factory_map()[name];
-  return std::unique_ptr<Base>(f->create());
+  return libmesh_make_unique<Base>(f->create());
```

Comment:
You'd think this was trivial, but it just broke a branch I'm working on!

```
../../../include/libmesh/auto_ptr.h(47): error: object of abstract class type "libMesh::Partitioner" is not allowed:
            function "libMesh::Partitioner::clone" is a pure virtual function
            function "libMesh::Partitioner::_do_partition" is a pure virtual function
    return std::unique_ptr<T>(new T(std::forward<Args>(args)...));
                                  ^
          detected during:
            instantiation of "std::unique_ptr<T, std::default_delete<T>> libMesh::make_unique<T,Args...>(Args &&...) [with T=libMesh::Partitioner, Args=<std::unique_ptr<libMesh::Partitioner, std::default_delete<libMesh::Partitioner>>>]" at line 149 of "../../../include/libmesh/factory.h"
            instantiation of "std::unique_ptr<Base, std::default_delete<Base>> libMesh::Factory<Base>::build(const std::string &) [with Base=libMesh::Partitioner]" at line 332 of "../../../../examples/adjoints/adjoints_ex1/adjoints_ex1.C"
```

Here ```f->create()``` isn't arguments to a new Base, it is itself a ```unique_ptr<Derived>``` that we're "casting" down to ```unique_ptr<Base>```.

Reply:
I'll add the fix in that branch.

I was *just* thinking that it was nice that we'd finally be getting some test coverage inside the library of the Factory class!

Reply:
Huh, yeah I can see the issue, but I wonder what compiler you are using that caught it when it had already passed all the CI testing?

Reply:
Not compiler, branch.  I guess we just never instantiated that method in GRINS or MOOSE?!  I know we didn't internally.

IIRC Factory is in use in FIN-S, though, so hopefully I'll finish testing this PR soon and we can get it merged before @benkirk has to step in with one of his quiet "they broke my stuff again" pushes to master.